### PR TITLE
Use consistent labels, remove additional settings, and copySmall icon LinkControl

### PR DESCRIPTION
**Pull request by richtabor**
## What?

Removes the "Open in new tab" control from the preview, originally applied in https://github.com/WordPress/gutenberg/pull/53566 as a stop-gap (now that https://github.com/WordPress/gutenberg/pull/57726 has landed, and https://github.com/WordPress/gutenberg/issues/50892 is on the way.)  

Also: 

- Changes tooltips to "Edit link", "Remove link", and "Copy link". 
- Adds new copySmall icon, as it fits better in this footprint. 

## Screenshots or screencast 
CleanShot 2024-01-24 at 08 06 15

### CODE REVIEW

**Comment by richtabor** [comment]
For reference, here's with the latest LinkControl changes + this PR: 


https://github.com/WordPress/gutenberg/assets/1813435/985a11ca-b562-4309-b93a-27e26eda7ee9


Compared to current WordPress 6.4: 

https://github.com/WordPress/gutenberg/assets/1813435/078fdd08-def4-493c-8b48-1b641110a7fd

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/link-preview.js:108]: I recommend we don't remove the tooltip. Why? Because histroically this has been required to allow users to see the full URL. 

This is necessary, for example, when you link to a file or an image which can have very long URLs.

Prior rationale for introducing this can be found in [this discussion](https://github.com/WordPress/gutenberg/pull/54920#issuecomment-1746384383).

In my opinion this change, as much as it seems like a nice "tidy", would be a regression.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (getdave) [inline packages/block-editor/src/components/link-control/link-preview.js]: I believe this is in order to remove the `Opens in new tab`.

I have two concerns. The first is technical and the second is based on principle.

Firstly, if we decide we do want to remove the `Opens in new tab` then we should restore this line and instead look in the main `index.js` file for where we _pass_ `additionalControl` to `<LinkPreview>`. The extension point should remain as is.

Secondly, if we are going to remove `Opens in new tab` from the preview _without_ first merging https://github.com/WordPress/gutenberg/pull/58279 then we should do this in it's own PR. 

Why? Because it's a contentious decision and _if_ we need to revert it, it would be easier for release leads to handle in a single PR which does one thing. 

I appreciate it might seem pedantic, but it will make a real different when we're in the full flow of pre-release flight checks 🙏 

Personally now the link preview remains open, I see an argument for removing the checkbox and instead requiring one additional click to change the setting. This is much better than it was previously, when the Link UI would close upon initial creation therefore requiring several clicks to reach the "edit" view again. This new UX is much smoother and so it's less of a concern to have the checkbox available on the preview.

File: packages/block-editor/src/components/link-control/style.scss
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:250]: Can I check this is legit from an a11y standpoint? Focus styles are extremely important to allow folks to understand where focus currently resides. It might be that we're just removing an _additional_ affordance and focus styles still remain in place. If so then we should probably add a comment to explain that, as this code could set a precedent for others in future to remove the remaining focus styles.

**Review by getdave — CHANGES_REQUESTED** [review]
Visually I like this. It's cleaner and neater.

I left a few comments with a few concerns which I think we should address before this is merged.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: The problem with having the tooltip is that focus is moved to it after link creation which reveals the tooltip and:
1. Covers the action buttons if it's a long URL
2. Obstructs the identical url text beneath it if it's a short URL

Tooltip on Link Preview covering the edit button

If the full url value is needed from the preview and we also don't want to display too much of the URL, then we could:
- Have the copy button tooltip include the full url, ie: "Copy: [URL value]"
- Have a button or tooltip icon next to truncated URLs that can be used to access the full value

File: packages/block-editor/src/components/link-control/style.scss
Comment (jeryj) [reply packages/block-editor/src/components/link-control/style.scss:250]: Looks like the focus ring is still there. I don't see any visual difference.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (richtabor) [reply packages/block-editor/src/components/link-control/link-preview.js]: > Secondly, if we are going to remove Opens in new tab from the preview without first merging https://github.com/WordPress/gutenberg/pull/58279 then we should do this in it's own PR.

I'm still not certain about #59279 to be honest — https://github.com/WordPress/gutenberg/pull/58279#issuecomment-1915841739

Comment (richtabor) [reply packages/block-editor/src/components/link-control/link-preview.js]: However, I do think removing it from the preview is warranted, accounting for the other optimizations we've done for link control. With this PR, enabling a link to open in a new tab is the same number of steps as it was previously, but much less stressing on the user experience, as the link popover doesn't close and require you to re-open it. It feels _lighter_, because it is. 

Adding to the preview was always a stopgap, awaiting the experience unlocked by the combination of https://github.com/WordPress/gutenberg/pull/57986 and https://github.com/WordPress/gutenberg/pull/57726.

Comment (getdave) [reply packages/block-editor/src/components/link-control/link-preview.js]: > feels lighter, because it is.

Based on my experience I would tend to agree.

I would still recommend we do this bit as a _separate_ PR 🙏

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (getdave) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: Understood. Thanks for illustrating that 👍 

There is perhaps an argument around the copy button. How about we move the tooltip to the `Copy` button? 

I'm open to any suggestion really.

I wanted to make sure we understood that having a tooltip was an intentional UX decision to alleviate real-world user feedback and so we shouldn't just remove it without providing a solid alternative.

I appreciate this was easy to miss though 🙇

Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: If we do use "Copy: [URL Value]", I believe we need to set the aria-label as only "Copy" so that the button has a consistent accessible name.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js]: I removed this from this PR. It can be done in a follow-up.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (getdave) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: I don't think we implemented the suggestion here https://github.com/WordPress/gutenberg/pull/58183/files#r1471501443. Once done I'll dismiss my blocking review 🙏

Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: I'm not sure about my comment's validity there. I think it might be fine for it to be "Copy: [URL Value]."

Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: I'll add the URL to the Copy tooltip and see how it feels.

Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: https://github.com/WordPress/gutenberg/pull/58183/commits/0a6d92c6e12eb5bfe75c355e6c22e96d4d7d18fb

Comment (getdave) [reply packages/block-editor/src/components/link-control/link-preview.js:108]: I tried using `sprintf` which I hope makes it clearer and is still legit. Not 100% sure though 😓

**Review by getdave — APPROVED** [review]
Thank you to @jeryj and @richtabor for resolving my concerns:

- breaking removal of `Opens in new tab` checkbox into a seperate PR
- ensuring we still have a means to view the full URL (now via Copy button)

Screen Shot 2024-02-07 at 10 51 40 (1)

LGTM 🚀

**Review by scruffian — APPROVED** [review]
LGTM

**Pull request merged**
